src/sentry/static/sentry/app/views/performance/transactionSummary/transactionList.tsx

```diff
@@ -146,6 +148,83 @@ class TransactionTable extends React.PureComponent<Props> {
     });
   };
 
+  handleCellAction = (column: TableColumn<React.ReactText>) => {
+    return (action: Actions, value: React.ReactText) => {
+      const {eventView, location} = this.props;
+
+      let nextLocationQuery: Location['query'] = {};
+      const searchConditions = tokenizeSearch(eventView.query);
+
+      // remove any event.type queries since it is implied to apply to only transactions
+      delete searchConditions['event.type'];
+
+      // no need to include transaction as its already in the query params
+      delete searchConditions.transaction;
+
+      switch (action) {
+        case Actions.ADD: {
+          // Remove exclusion if it exists.
+          delete searchConditions[`!${column.name}`];
+          searchConditions[column.name] = [`${value}`];
+
+          nextLocationQuery = {
+            query: stringifyQueryObject(searchConditions),
+          };
+          break;
+        }
+        case Actions.EXCLUDE: {
+          // Remove positive if it exists.
+          delete searchConditions[column.name];
+          // Negations should stack up.
+          const negation = `!${column.name}`;
+          if (!searchConditions.hasOwnProperty(negation)) {
+            searchConditions[negation] = [];
+          }
+          searchConditions[negation].push(`${value}`);
```

Comment:
We now have a few different implementations of this kind of logic. I wonder if we could encapsulate it better in the new Query object that Evan built? 🤔

Reply:
Let me see if there's some nice way to consolidate all of them together.